src/components/box/box.test.js

```diff
@@ -0,0 +1,666 @@
+      "should verify Box opacity is %s",
+      (opacity) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" opacity={opacity} />
+        );
+
+        getDataElementByValue("box").should("have.css", "opacity", opacity);
+      }
+    );
+
+    it.each(widthConstants)(
+      "should verify Box width is %s when prop is passed as a percentage",
+      (width, percentage) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" width={percentage} />
+        );
+
+        getDataElementByValue("box")
+          .should("have.attr", "width", percentage)
+          .and("have.css", "width", width);
+      }
+    );
+
+    it.each(widthConstants)(
+      "should verify Box width is %s when prop is passed as a number",
+      (width, percentage, number) => {
+        CypressMountWithProviders(<BoxComponent bg="primary" width={number} />);
+
+        getDataElementByValue("box")
+          .should("have.attr", "width", number)
+          .and("have.css", "width", width);
+      }
+    );
+
+    it.each(widthConstants)(
+      "should verify Box width is %s when prop is passed as a string",
+      (width) => {
+        CypressMountWithProviders(<BoxComponent bg="primary" width={width} />);
+
+        getDataElementByValue("box")
+          .should("have.attr", "width", width)
+          .and("have.css", "width", width);
+      }
+    );
+
+    it.each(heightConstants)(
+      "should verify Box height is %s when prop is passed as a number",
+      (height, number) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" height={number} />
+        );
+
+        getDataElementByValue("box")
+          .should("have.attr", "height", number)
+          .and("have.css", "height", height);
+      }
+    );
+
+    it.each(heightConstants)(
+      "should verify Box height is %s when prop is passed as a string",
+      (height) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" height={height} />
+        );
+
+        getDataElementByValue("box")
+          .should("have.attr", "height", height)
+          .and("have.css", "height", height);
+      }
+    );
+
+    it.each(sizeConstants)(
+      "should verify Box height and width are %s when Size prop is passed as a number",
+      (pixels, percentage, number) => {
+        CypressMountWithProviders(<BoxComponent size={number} />);
+
+        getDataElementByValue("box")
+          .should("have.css", "width", pixels)
+          .and("have.css", "height", pixels);
+      }
+    );
+
+    it.each(sizeConstants)(
+      "should verify Box height and width are %s when Size prop is passed as a string",
+      (pixels) => {
+        CypressMountWithProviders(<BoxComponent size={pixels} />);
+
+        getDataElementByValue("box")
+          .should("have.css", "width", pixels)
+          .and("have.css", "height", pixels);
+      }
+    );
+
+    it.each(["contents", "list-item", "none"])(
```

Comment:
```suggestion
    it.each(["block", "inline-block", "flex", "contents", "list-item", "none"])(
```
It's good you're trying to cover all the possible values for these CSS properties.

After reading through the file, I agree with what has been said before about our documentation for these surfaced CSS properties. I think we can amend the wording for each prop description - mentioning if it refers to a CSS property and not a custom React prop we've created - perhaps including a link to the relevant MDN docs on that CSS property.